examples/hif2a/fit_to_multiple_rbfes.py

```diff
@@ -120,9 +125,36 @@ def _blew_up(results: List[SimulationResult]) -> bool:
     # TODO: adjust this threshold a bit, move reliability calculations into fe/estimator.py or fe/model.py
     return np.isnan(du_dls).any() or (du_dls.std(1).max() > 1000)
 
-def loss_fxn(model, ff_params, mol_a, mol_b, core, label_ddG):
-    pred_ddG, stage_results = model.predict(ff_params, mol_a, mol_b, core)
-    return pseudo_huber_loss(pred_ddG - label_ddG), stage_results
+# def loss_fxn(model, ff_params, mol_a, mol_b, core, label_ddG):
+#     pred_ddG, stage_results = model.predict(ff_params, mol_a, mol_b, core)
+#     return pseudo_huber_loss(pred_ddG - label_ddG), stage_results
+
+def loss_fxn(ff_params, batch: List[Tuple[RelativeFreeEnergy, RBFEModel]]):
+    index = RBFETransformIndex()
+    index.build([edge[0] for edge in batch])
+    indices = []
+    all_results = []
+    preds = []
+    for rfe, model in batch:
+        indices.append(list(index.get_transform_indices(rfe)))
+        pred_ddG, stage_results = model.predict(ff_params, rfe.mol_a, rfe.mol_b, rfe.core)
+        all_results.extend(list(stage_results))
+        preds.append(pred_ddG)
+    # If the edges in the batch are within a cycle, correct the ddGs
+    indices = jnp.asarray(indices)
+    ind_l, ind_r = indices.T
+    try:
+        layer = construct_mle_layer(len(index), indices)
+        corrected_dg = layer(jnp.asarray(preds))
+        cycle_corrected_rbfes = corrected_dg[ind_r] - corrected_dg[ind_l]
+    except DisconnectedEdgesError:
+        # Provided non connected graph, ie its not a cycle
```

Comment:
Nit: Should we modify this comment slightly to distinguish between "connected" and "contains cycles"? (a minimum spanning tree for example would be connected but not contain any cycles)